### DISCUSSION TITLE
Function return signatures

### DISCUSSION THREAD

**Discussion by tarkatronic** [underdog-tech/vulnbot#23] [General]
Right now a lot of functions in the code will detect error scenarios, or missing data, and return empty data structures. For example here: https://github.com/underdog-tech/vulnbot/blob/2f14fd6c882f7ac5feffcd8af16e541a299f174a/config/config.go#L68

I get the sense that this isn't really a best practice in Go, and it also leads to some code that just puts me on edge a bit. For example, to handle that return value we have to have this: https://github.com/underdog-tech/vulnbot/blob/2f14fd6c882f7ac5feffcd8af16e541a299f174a/internal/scan.go#L229

And that seems not great.

I suspect that, instead, we should change the return signature to `(TeamConfig, Error)` and return an error when no config is found. This seems to be far more the norm in Go code. Then to handle it, there's the standard
```go
config, err := config.GetTeamConfigBySlug(...)
if err != nil {
    // ...
}
```

Does anybody have thoughts on this?

**Comment by tarkatronic** [comment]
@JoseAngel1196 @developerDemetri I figure you two might have input here.

**Reply by developerDemetri** [reply]
Agreed, it seems like returning `(nil, ConfigNotFoundError)` is more in line with Go

**Reply by JoseAngel1196** [reply]
Yup! I like this idea, this current code seems too Pythonic to me, personally 😅. It is a common practice to use the "multiple return values" to indicate the result and error status of a function as you said. 💯

**Comment by tarkatronic** [comment]
Fixed via #24
